plugins/rich-editor/src/scripts/blots/LinkEmbed.js

```diff
@@ -0,0 +1,148 @@
+            source.classList.add('meta');
+            source.innerHTML = data.source;
+        }
+
+        let linkImage = false;
+        if (data.linkImage) {
+            linkImage = document.createElement('div');
+            linkImage.classList.add('embedLink-image');
+            linkImage.setAttribute('aria-hidden', 'true');
+            linkImage.setAttribute('style', 'background-image: url("' + data.linkImage + '");');
+        }
+
+        let userName = false;
+        if (data.userName) {
+            userName = document.createElement('span');
+            userName.classList.add('embedLink-userName');
+            userName.innerHTML = data.userName;
+        }
+
+        let dateTime = false;
+        if (data.timestamp) {
+            dateTime = document.createElement('time');
+            dateTime.classList.add('embedLink-dateTime');
+            dateTime.classList.add('meta');
+            dateTime.setAttribute('datetime', data.timestamp);
+            dateTime.innerHTML = data.humanTime;
+        }
+
+        let title = false;
+        if (data.name) {
+            title = document.createElement('h3');
+            title.classList.add('embedLink-title');
+            title.innerHTML = data.name;
+        }
+
+        const article = document.createElement('article');
+        article.classList.add('embedLink-body');
+
+        const main = document.createElement('div');
+        main.classList.add('embedLink-main');
+
+        const header = document.createElement('div');
+        header.classList.add('embedLink-header');
+
+        const excerpt = document.createElement('div');
+        excerpt.classList.add('embedLink-excerpt');
+        excerpt.innerHTML = data.excerpt;
+
+        // Assemble header
+        if (userPhoto) {
+            header.appendChild(userPhoto);
+        }
+
+        if (userName) {
+            header.appendChild(userName);
+        }
+
+        if (dateTime) {
+            header.appendChild(dateTime);
+        }
+
+        if (title) {
+            header.appendChild(title);
+        }
+
+        if (source) {
+            header.appendChild(source);
+        }
+
+        // Assemble main
+        main.appendChild(header);
+        main.appendChild(excerpt);
+
+        // Assemble Article
+        if (linkImage) {
+            article.appendChild(linkImage);
+        }
+        article.appendChild(main);
+
+        // Assemble component
+        node.appendChild(article);
+
+        setData(node, "data", data);
+        return node;
+    }
+
+    static sanitize(url) {
+        return sanitize(url, this.PROTOCOL_WHITELIST) ? url : this.SANITIZED_URL;
+    }
+
+    static value(node) {
+        return getData(node, "data");
+    }
+}
+
+function getLineHeight(node){
+    let temp = document.createElement(node.nodeName);
+    temp.setAttribute("style", "margin:0px;padding:0px;font-family:" + node.style.fontFamily + ";font-size:" + node.style.fontSize);
```

Comment:
This would be better if it used the style object instead of a string.

```js
temp.style = {
    ...node.style,
    margin: "0px",
    padding: "0px",
}
```